File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    public enum UserOrder { CreatedAt, LastActive }

    public class UserParams : PaginationParams
    {
        public string CurrentUserName { get; set; }
        public string Gender { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 150;
        public UserOrder OrderBy { get; set; } = UserOrder.LastActive;
    }
}

File: AngularClient/src/app/models/userParams.ts
import { User } from './user'

export type UserOrder = 'CreatedAt' | 'LastActive'

export type Gender = 'female' | 'male'

export class UserParams {
    pageNumber = 1
    pageSize = 5
    minAge = 18
    maxAge = 99
    gender: Gender
    orderBy: UserOrder = 'LastActive'

    constructor(user: User) {
        this.gender = user.gender === 'female' ? 'male' : 'female'
    }
}
